tensorflow/core/kernels/split_lib_gpu.cu.cc

```diff
@@ -124,7 +124,11 @@ __global__ void split_v_kernel(const T* input_ptr,
   int num_outputs = output_ptr_data.size;
 
   // verbose declaration needed due to template
+#if GOOGLE_CUDA
```

Comment:
Have you considered introducing a wrapper for this?

Reply:
I had not until now.

would something the following lines work

```
<in tensorflow/core/gpu_kernel_helper.h>

#if GOOGLE_CUDA
  #define GPU_DYNAMIC_SHARED_MEM_DECL(ALIGN, TYPE, NAME)     extern __shared__ __align__(ALIGN) TYPE NAME[]
#elif TENSORFLOW_USE_ROCM
  #define GPU_DYNAMIC_SHARED_MEM_DECL(ALIGN, TYPE, NAME)     HIP_DYNAMIC_SHARED(TYPE, NAME)
#endif 


<and then in the file where we need the shared memory decl, like right here>

GPU_DYNAMIC_SHARED_MEM_DECL(sizeof(T), unsigned char, smem);

```

There are 8 instances of HIP_DYNAMIC_SHARED in our fork...so this should not be a major change.

Assuming the above change is the route we want to go, can we track it in a different PR, and leave the change here as is?

thanks

Reply:
@chsigg , gentle ping ... let me know if you are okay with the proposal above, and whether we can address it in a separate PR.

Reply:
cool. expect a PR for it sometime next week.

also this PR should be good to go now.

thanks again